### PR TITLE
updates to support library packaging

### PR DESCRIPTION
**Pull request by wbgilmartin**
Made updates to TensileCreateLibrary to enable the packaging of the new client library code objects into separate architectures as sub-directories within the Tensile/library directory. 

The legacy library generation is still preserved, the packaging is enabled with the --package-library switch.

### CODE REVIEW

File: Tensile/TensileCreateLibrary.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 from .KernelWriterSource import KernelWriterSource
 from .SolutionStructs import Solution
 from .SolutionWriter import SolutionWriter
+from .SolutionLibrary import MasterSolutionLibrary
```
Comment (amdkila) [inline Tensile/TensileCreateLibrary.py]: This is unused and is causing a lint error. Can this line be removed?

File: Tensile/Source/TensileCreateLibrary.cmake [hunk 2]
```diff
@@ -32,7 +32,13 @@ function(TensileCreateLibraryCmake
     Tensile_MERGE_FILES
     Tensile_SHORT_FILE_NAMES
     Tensile_LIBRARY_PRINT_DEBUG )
-
+# make Tensile_PACKAGE_LIBRARY and optional parameter
```
Comment (sdquiring) [inline Tensile/Source/TensileCreateLibrary.cmake]: Can you update the new version of this function as well?

File: Tensile/TensileCreateLibrary.py [hunk 3]
```diff
@@ -1000,13 +1015,20 @@ def TensileCreateLibrary():
...
 
+  masterLibraries = {}
+  fullMasterLibriary = None
   for logic in Utils.tqdm(libraries, "Processing logic data"):
     (scheduleName, deviceNames, problemType, solutionsForSchedule, \
-       indexOrder, exactLogic, rangeLogic, newLibrary) = logic
+       indexOrder, exactLogic, rangeLogic, newLibrary, architectureName) = logic
+
+    if not globalParameters["PackageLibrary"]:
+      if fullMasterLibriary is None:
```
Comment (sdquiring) [inline Tensile/TensileCreateLibrary.py]: Spelling

File: Tensile/TensileCreateLibrary.py [hunk 4]
```diff
@@ -161,26 +162,37 @@ def buildSourceCodeObjectFile(CxxCompiler, outputPath, kernelFile):
...
 
-    return destCOs
+    return destCosList
 
 def buildSourceCodeObjectFiles(CxxCompiler, kernelFiles, outputPath):
     args = zip(itertools.repeat(CxxCompiler), itertools.repeat(outputPath), kernelFiles)
 
     coFiles = Common.ParallelMap(buildSourceCodeObjectFile, args, "Compiling source kernels",
-                                 method=lambda x: x.starmap)
+                                 method=lambda x: x.starmap, enable=False)
```
Comment (sdquiring) [inline Tensile/TensileCreateLibrary.py]: Does this not work with parallelization?

File: Tensile/TensileCreateLibrary.py [hunk 5]
```diff
@@ -301,7 +312,9 @@ def writeSolutionsAndKernels(outputPath, CxxCompiler, problemTypes, solutions, k
   prepAsm()
 
   kIter = zip(kernels, itertools.repeat(kernelWriterSource), itertools.repeat(kernelWriterAssembly))
-  results = Common.ParallelMap(processKernelSource, kIter, "Generating kernels", method=lambda x: x.starmap)
+  results = Common.ParallelMap(processKernelSource, kIter, "Generating kernels", method=lambda x: x.starmap, enable=False)
```
Comment (sdquiring) [inline Tensile/TensileCreateLibrary.py]: `enable=False` left in

File: Tensile/TensileCreateLibrary.py (hunk 4 above)
Comment (wbgilmartin) [reply Tensile/TensileCreateLibrary.py]: I added this for debugging and forgot to remove it.

File: Tensile/Source/TensileCreateLibrary.cmake (hunk 2 above)
Comment (wbgilmartin) [reply Tensile/Source/TensileCreateLibrary.cmake]: Not sure what you are asking here is there another version of this function?

Comment (sdquiring) [reply Tensile/Source/TensileCreateLibrary.cmake]: The newer version is in [TensileConfig.cmake](https://github.com/ROCmSoftwarePlatform/Tensile/blob/a89467f8e7627782acf5c5e24e82095cea39aa9d/Tensile/cmake/TensileConfig.cmake#L68).

**Pull request merged**